bugbug/repository.py

```diff
@@ -216,6 +222,25 @@ def download_commits(repo_dir, date_from):
     r.raise_for_status()
     COMPONENTS = r.json()
 
+    components_touched = {}
+
+    for commit in commits:
+        comp = set('::'.join(COMPONENTS[fl]) for fl in commit.files.decode('utf-8').split('|') if fl in COMPONENTS)
```

Comment:
Since we are repeating `'::'.join(COMPONENTS[fl])` all the time, maybe we should just do this once at the beginning by transforming COMPONENTS